### PR TITLE
Add GET single note, PUT, and DELETE APIs for notes, and more fixes

### PR DESCRIPTION
**Pull request by elveskevtar**

### CODE REVIEW

File: src/api/note_service.py [hunk 1]
```diff
@@ -1,12 +1,42 @@
...
+    return NotesDB.get_notes_for_user(author_id)
 
 
-def post_note(
+def get_note_by_id(author_id: int, note_id: int) -> Note | None:
+    note = NotesDB.get_note_by_id(note_id)
+    if note is None:
+        return None
+
+    print(type(note_id))
```
Comment (elveskevtar) [inline src/api/note_service.py:15]: Ooops this should not be here, was for testing

File: src/api/note_service.py [hunk 2]
```diff
@@ -1,12 +1,42 @@
...
+    if note.is_public or note.author_id == author_id:
+        return note
+
+    return None
+
+    print(note.author_id)
+    return note if note.is_public or note.author_id == author_id else None
```
Comment (wbc2025) [inline src/api/note_service.py:22]: These last two lines are unreachable because of the return statement above

File: src/db/notes.py [hunk 3]
```diff
@@ -25,7 +38,51 @@ def create_note(
...
+
+
+def update_note(
+    note_id: int,
+    note_title: str,
+    note_text: str,
+    author_id: int,
+    is_public: bool = False,
+) -> Note | None:
+    with get_db() as db:
```
Comment (wbc2025) [inline src/db/notes.py:55]: If the user only wants to update the title, then potentially they don't include the note_text and is_public, so those default in app.py to None and False, respectively. The default values then get passed all the way to the NoteDB, where they aren't checked and will update the note, which would unintentially change fields that the user didn't intend.

**Review by wbc2025 — CHANGES_REQUESTED** [review]

File: src/api/note_service.py (hunk 2 above)
Comment (elveskevtar) [reply src/api/note_service.py:22]: No idea how this happened lol, let me fix it

File: src/db/notes.py (hunk 3 above)
Comment (elveskevtar) [reply src/db/notes.py:55]: Right now, the only API we have for updating is a PUT request (normally used for updated/replacing a whole object).

I understand what you're saying from a DAO layer though. I will updates this to update only provided fields and that way we can have something like a PATCH request in the future.

**Review by wbc2025 — APPROVED** [review]

**Pull request merged**